composeApp/src/commonMain/kotlin/com/github/familyvault/ui/screens/main/familyGroupSettings/ChangeFamilyGroupScreen.kt

```diff
@@ -38,23 +43,45 @@ import org.jetbrains.compose.resources.stringResource
 import org.koin.compose.koinInject
 
 class ChangeFamilyGroupScreen : Screen {
+    private val usernameByContextId = mutableStateMapOf<String, String>()
+    private val familyGroups = mutableStateListOf<FamilyGroup>()
+    
     @Composable
     override fun Content() {
         val savedFamilyGroupsService = koinInject<ISavedFamilyGroupsService>()
         val familyGroupSessionService = koinInject<IFamilyGroupSessionService>()
+        val familyGroupService = koinInject<IFamilyGroupService>()
         val navigator = LocalNavigator.currentOrThrow
 
         var isLoading by mutableStateOf(false)
         var isChangingFamilyGroup by mutableStateOf(false)
-        val familyGroups = mutableStateListOf<FamilyGroup>()
         val currentContextId =
             familyGroupSessionService.takeIf { it.isSessionAssigned() }?.getContextId()
         val coroutineScope = rememberCoroutineScope()
 
-
         LaunchedEffect(Unit) {
             isLoading = true
+            familyGroups.clear()
             familyGroups.addAll(savedFamilyGroupsService.getAllSavedFamilyGroups())
+
+            for (familyGroup in familyGroups) {
+                try {
+
+                    familyGroupSessionService.disconnect()
+                    familyGroupSessionService.assignSession(
+                        familyGroupCredential = savedFamilyGroupsService.getSavedFamilyGroupCredentialByContextId(
+                            familyGroup.contextId
+                        )
+                    )
+                    familyGroupSessionService.connect()
+
+                    val myData = familyGroupService.retrieveMyFamilyMemberData()
+                    usernameByContextId[familyGroup.contextId] = myData.firstname
+                } catch (e: Exception) {
+                    continue
+                }
+            }
```

Comment:
Ten kod jest nieoptymalny i dodatkowo **wprowadza krytyczny błąd**. Gdy wejdzie się w listę rodzin to automatycznie zostaniemy przeniesieni do ostatniej grupy rodzinnej.

Odtworzenie:
1. Stwórz grupe rodzinną A
2. Stwórz grupe rodzinną B
3. Przejdź do grupy rodzinnej A
4. Wejdź do ekranu grupy rodzinnej, nie wybieraj nic
5. Jesteś w grupie B, mimo że nic nie wybrałeś

 Imie i nazwisko nie powinny być w taki sposób pobierane i powinny być zapisywanie w `FamilyGroupCredential`. 
![image](https://github.com/user-attachments/assets/35a1620a-c0ae-4ca5-86bf-9401c891cdbb)